roles/azure_manage_postgresqlflex/tasks/create.yml

```diff
@@ -0,0 +1,83 @@
+---
+- name: Check that Azure Region is known
+  ansible.builtin.fail:
+    msg: Azure location must be defined as azure_manage_postgresqlflex_region
+  when:
+    - rg_info.resourcegroups | length == 0
+    - azure_manage_postgresqlflex_region is not defined
+
+- name: Check Azure PostgreSQL admin user
+  ansible.builtin.fail:
+    msg: Azure PostgreSQL Server admin user must be defined as azure_manage_postgresqlflex_admin_username
+  when:
+    - server_info.servers | length == 0
+    - azure_manage_postgresqlflex_admin_username is not defined
+
+- name: Check Azure PostgreSQL admin password
+  ansible.builtin.fail:
+    msg: Azure PostgreSQL Server admin password must be defined as azure_manage_postgresqlflex_admin_password
+  when:
+    - server_info.servers | length == 0
+    - azure_manage_postgresqlflex_admin_password is not defined
+
+- name: Create resource group if it doesn't already exists
+  ansible.builtin.include_role:
+    name: cloud.azure_ops.azure_manage_resource_group
+  vars:
+    azure_manage_resource_group_operation: create
+    azure_manage_resource_group_name: "{{ azure_manage_postgresqlflex_resource_group }}"
+    azure_manage_resource_group_region: "{{ azure_manage_postgresqlflex_region }}"
+    azure_manage_resource_group_tags: "{{ azure_manage_postgresqlflex_tags | default({}) }}"
+  when: rg_info.resourcegroups | length == 0
```

Comment:
how is this code supposed to work? you have an assert at the top that fails when `rg_info.resourcegroups | length == 0` i.e. the same condition from this block